pkg/process/monitor/process_monitor.go

```diff
@@ -347,16 +349,28 @@ func (pm *ProcessMonitor) mainEventLoop() {
 //  2. Initializes the netlink process monitor.
 //  2. Run the main event loop in a goroutine.
 //  4. Scans already running processes and call the Exec callbacks on them.
-func (pm *ProcessMonitor) Initialize() error {
+func (pm *ProcessMonitor) Initialize(useEventStream bool) error {
 	var initErr error
 	pm.initOnce.Do(
 		func() {
-			log.Info("initializing process monitor")
+			var method string
+			if useEventStream {
+				method = "event stream"
+			} else {
+				method = "netlink"
+			}
+			log.Infof("initializing process monitor (%s)", method)
 			pm.tel = newProcessMonitorTelemetry()
-			pm.done = make(chan struct{})
+
 			pm.initCallbackRunner()
 
 			pm.processMonitorWG.Add(1)
+
+			if useEventStream {
+				return
+			}
+
+			pm.done = make(chan struct{})
```

Comment:
I'm afraid the changes here are going to create deadlocks and issues during shutdown (not deterministically, but during races)

Reply:
`Stop` seems to be overloaded with different sync primitives. Any chance we can clean that up?

Reply:
Would moving this line back to where it was solve the issue for the time being?